.github/workflows/esp32.yml

```
@@ -4,87 +4,148 @@ name: CI
+          python-version: '3.11'
+      - name: Install LittleFS Tool
+        run: pip install littlefs-python
+      - name: Create LittleFS Image
+        run: littlefs-python create $(pwd)/data littlefs.bin -v --fs-size=0x20000 --name-max=64 --block-size=4096
+      - name: Archive LittleFS image
         uses: actions/upload-artifact@v4
         with:
-          name: esp32s3-app-firmware
-          path: .pio/build/s3/firmware.bin
+          name: littlefs-binary
+          path: littlefs.bin
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)